fud/fud/stages/xilinx/execution.py

```diff
@@ -75,21 +60,56 @@ def run(xclbin: SourceType.Path) -> SourceType.String:
                         f"user_pre_sim_script={new_dir.name}/pre_sim.tcl\n"
                     )
                     xrt_ini_config.append(
-                        f"user_post_sim_script={new_dir.name}/pre_sim.tcl\n"
+                        f"user_post_sim_script={new_dir.name}/post_sim.tcl\n"
                     )
                 f.writelines(xrt_ini_config)
 
             # Extra Tcl scripts to produce a VCD waveform dump.
             if waveform:
                 with open("pre_sim.tcl", "w") as f:
-                    f.writelines([
-                        "open_vcd\n",
-                        "log_vcd *\n",
-                    ])
+                    f.writelines(
+                        [
+                            "open_vcd\n",
+                            "log_vcd *\n",
+                        ]
+                    )
                 with open("post_sim.tcl", "w") as f:
-                    f.writelines([
-                        "close_vcd\n",
-                    ])
+                    f.writelines(
+                        [
+                            "close_vcd\n",
+                        ]
+                    )
+
+            os.environ["XRT_INI_PATH"] = f"{new_dir.name}/xrt.ini"
+
+        @builder.step()
+        def import_libs():
+            """Import optional libraries"""
+            try:
+                from fud.stages.xilinx import fud_pynq_script
+
+                self.pynq_script = fud_pynq_script
+            except ImportError:
+                raise errors.RemoteLibsNotInstalled
+
+        @builder.step()
+        def run(xclbin: SourceType.Path) -> SourceType.String:
+            """Run the xclbin with datafile"""
+
+            if data_path is None:
+                raise errors.MissingDynamicConfiguration("fpga.data")
+            # Go up a directory due to config going down
+            abs_data_path = Path("../" + data_path).resolve()
+            xclbin = Path("..") / xclbin
+            abs_xclbin_path = xclbin.resolve()
+
+            # Import optional libraries
+            try:
+                from fud.stages.xilinx import fud_pynq_script
+
+                self.pynq_script = fud_pynq_script
+            except ImportError:
+                raise errors.RemoteLibsNotInstalled
```

Comment:
Is this duplicated import stanza necessary? (It seems like it should already be good to go after the `import_libs` stage completes.